.github/workflows/test.yml

```diff
@@ -75,17 +85,30 @@ jobs:
           target: x86_64-pc-windows-gnu
           default: true
           components: rustfmt, clippy
+      - name: Add i686 target
+        run: rustup target add i686-pc-windows-gnu
       - name: Build
         run: |
           echo "C:/msys64/usr/bin" >> $GITHUB_PATH
           echo "C:/msys64/mingw64/bin" >> $GITHUB_PATH
-          cargo build
+          cargo build -vv
+        env:
+          LIBCLANG_PATH: C:/msys64/mingw64/bin
+      - name: Build i686 version
+        run: |
+          echo "C:/msys64/usr/bin" >> $GITHUB_PATH
+          echo "C:/msys64/mingw64/bin" >> $GITHUB_PATH
+          set MSYSTEM=MINGW32
+          cargo build -vv --target i686-pc-windows-gnu
         env:
           LIBCLANG_PATH: C:/msys64/mingw64/bin
       - name: Run tests
         run: |
           echo "C:/msys64/usr/bin" >> $GITHUB_PATH
           echo "C:/msys64/mingw64/bin" >> $GITHUB_PATH
-          cargo test -- --nocapture --test-threads=1
+          cargo test --target x86_64-pc-windows-gnu -- --nocapture --test-threads=1
+          echo "C:/msys64/mingw32/bin" >> $GITHUB_PATH
```

Comment:
[GitHub Actions doc](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#adding-a-system-path) says:

> Prepends a directory to the system `PATH` variable for all subsequent actions in the current job. The currently running action cannot access the new path variable.

So, I think this doesn't work. Maybe we can just set `PATH` envvar by `$ENV:Path="...;"+$ENV:Path`? (Is the shell PowerShell here, right...?)

Reply:
Ah, maybe I just have to set it once during the setup. I'll try it.

Reply:
So that helped, actually. Now the test compiles but then crashes. :-) It also means all these `echo` statements were not needed.

Reply:
The error is 0xc000007b `INVALID_IMAGE_FORMAT`, which is apparently a 32-bit/64-bit mismatch: https://social.msdn.microsoft.com/Forums/vstudio/en-US/c1147429-95f8-47e7-a532-f3960118ce11/x86-application-gives-invalidimageformat-0xc000007b-error-on-windows-7-64-bit?forum=clr

So something still isn't quite right with the setup.